### DISCUSSION TITLE
Collapsible Tree View for 'Json' Directus Type

### DISCUSSION THREAD

**Discussion by cmjdev** [directus/directus#8754] [Feature Requests]
As title states, Directus is very flexible but we are doing recursive templating in JSON and it would be nice to be able to display a JSON field type in a tree view (at least)

**Comment by rijkvanzanten** [comment]
When you say "JSON tree view", do you mean like a graphical JSON editor like https://jsoneditoronline.org ?

**Reply by cmjdev** [reply]
I was hoping for just a collapsible option in the JSON display box.. But that graphical editor is slick too!

**Reply by rijkvanzanten** [reply]
Collapsible option? Do you have any examples? 🙂

**Reply by azrikahar** [reply]
@cmjdev do you by any chance mean something like this? https://codemirror.net/demo/folding.html

**Reply by cmjdev** [reply]
> @cmjdev do you by any chance mean something like this? https://codemirror.net/demo/folding.html

Yes.. "Code Folding" is what I should have stated.

**Reply by rijkvanzanten** [reply]
> > @cmjdev do you by any chance mean something like this? https://codemirror.net/demo/folding.html
> 
> 
> 
> Yes.. "Code Folding" is what I should have stated.

We currently use code mirror for the json interface, so that option should be fairly easy to enable! Wanna open a PR for that?

**Reply by cmjdev** [reply]
Yep, I'll put that together this afternoon.

**Reply by azrikahar** [reply]
> Yes.. "Code Folding" is what I should have stated.

No worries. I was just hoping that is indeed what you want, since it should be easier to add as we're using codemirror already as Rijk mentioned ;)
